src/airflow_tools/providers/http_to_data_lake/operators/http_to_data_lake.py

```diff
@@ -190,6 +196,20 @@ def _response_filter(self, response) -> BytesIO:
                     )
                 return list_to_jsonl(self.data, self.compression)
 
+            case 'xml':
+                if not isinstance(self.data, str):
```

Comment:
Is this case ever possible if the if statement above is correct? We'll always have `response.text` at this point

Reply:
I think if we change the initial if statements this should not be needed.